src/sentry/api/serializers/rest_framework/rule.py

```diff
@@ -38,6 +39,7 @@ def from_native(self, data):
 
 class RuleSerializer(serializers.Serializer):
     name = serializers.CharField(max_length=64)
+    environment = serializers.CharField(max_length=64, required=False)
```

Comment:
This might be worth noting: I'm exposing these as strings (the environment name) rather than passing an environment ID around for consistency with how we address environments everywhere else.

This also means that if you typo the environment name when creating a rule, you'd end up creating a new environment with the bad name and not get an error. (The UI might make this less of a practical concern based on what form controls are used here.) If we'd rather not do that, I can make it so that this will error if the environment doesn't already exist.

Reply:
I was about to say "We definitely shouldn't let this create a new environment". But I can see someone setting a rule for an environment that hasn't been used yet intentionally?

Also, worth noting that if we _are_ creating environments here, this is bypassing other validation of what a valid name is, besides simply the length. So we should either hard error on this (I'd probably vote this for now) then unify the creation the same way we do in the pipeline to make sure we cover the same cases. Like, when we introduce normalizing it down to lowercase, etc. Or checking for valid characters.

Reply:
Validation is a good point. I'm inclined just to change this to limit the options to environments that have already been created for simplicity's sake. (My assumption would that this would be an autocomplete dropdown in the UI, but that's just an assumption.)

Reply:
Yeah, agreed. I think we just do that for now and deal with it later if we do want to create from here.

Reply:
When this makes it's way into the interface would we be able to have autocomplete on the currently available environments?